### ISSUE TITLE
Adding accessibility tags to a label seems to cause the renderer to need more space 

### ISSUE THREAD

**Issue by raymondjstone** [xamarin/Xamarin.Forms#2580]
### Description

Adding accessibility tags to a label seems to cause the renderer to need more space for the label when it is in stacklayouts (Horizontal orientated ones especially). Given accessibility tags are not visible there should be no requirement for more space surely

### Steps to Reproduce

1. create a  series of labels such they they use up the screen width on a horizontal stacklayout
2. add accessibility tags to the labels
3. 

### Expected Behavior
Labels should show the same with and without the tags
### Actual Behavior
Labels seem to 'need' more space on the screen and affect the layour
### Basic Information

- Version with issue: all possibly, am on the current version
- Last known good version:  none
- IDE: VS 2017 15.6.7
- Platform Target Frameworks:4.9.0.763
  - iOS:  11.9.1.24
  - Android: 8.2.0.16 
  - UWP:  n/a
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

**Comment by raymondjstone** [comment]
You are not setting the name nor the helptext fields and I am also setting the inaccessibletree flag to true (not sure if that is the default for labels or not)

**Comment by raymondjstone** [comment]
ok, some info

Firstly I mostly test on Android (easier when using VS2017 on windows), so by the time I deploy to IOS I had fixed the issue so issue is certainly on Android but I have no reason to believe it is not on IOS as well

I am only using C# not XAML for this, and all objects go through the same code (unless data bound), labels cause issues as described above but when I switched to using images but with the exact same accessibility settings the layout was fine.

The settings used are as following (labeledby is null in these cases, and the sanitize function currently does nothing except adjust the text to correct some bad pronunciations, intree is a bool that is true, name and helptext are strings )

            AutomationProperties.SetIsInAccessibleTree(o, inTree);
            if (inTree)
            {
                AutomationProperties.SetName(o, sanitize(name));
                AutomationProperties.SetHelpText(o, sanitize(helptext));
                if (labeledby != null)
                {
                    AutomationProperties.SetLabeledBy(o, labeledby);
                }
            }

**Comment by raymondjstone** [comment]
oh and please note you should have something visible below the accessible stacklayout as well, that way you should see it move down further than it should be

**Comment by hartez** [comment]
It looks like the problem only exists on Android, and setting `AutomationProperties.HelpText` and `AutomationProperties.Name` will trigger it. If you run the attached repro project on Android, the two rows of Labels do not line up; on the other platforms they do.

[_2580 Repro.zip](https://github.com/xamarin/Xamarin.Forms/files/1970970/_2580.Repro.zip)

**Comment by raymondjstone** [comment]
Thanks for the update and more accurate diagnosis

**Comment by pamela032709** [comment]
This is not working for me at all. only happen on android . 
ex:   <Button Text="Rec"
                        AutomationId="Record"
                        AutomationProperties.Name="Record"
                        AutomationProperties.IsInAccessibleTree="true"
                        Command="{Binding Record}"
                        x:Name="Record"
                        Style="{StaticResource RecordButton}"
                        TextColor="White"
                        CornerRadius="23" 
                        IsEnabled="{Binding AllowRecord}" 
                        AutomationProperties.HelpText="double tap to  start recording." 
                        HorizontalOptions="FillAndExpand" />
its size never expands on ios, but on Android, it takes the size of the help text label. is there any fix

**Comment by pamela032709** [comment]
ex: <Button Text="Rec"
                        AutomationId="Record"
                        AutomationProperties.Name="Record"
                        AutomationProperties.IsInAccessibleTree="true"
                        Command="{Binding Record}"
                        x:Name="Record"
                        Style="{StaticResource RecordButton}"
                        TextColor="White"
                        CornerRadius="23" 
                        IsEnabled="{Binding AllowRecord}" 
                        AutomationProperties.HelpText="double tap to  start recording." 
                        HorizontalOptions="FillAndExpand" />
it expands to the size of the help text string only on Android. never happen on ios.

**Comment by pamela032709** [comment]
> Is this occurring on all platforms or a specific platform? Could you post an example layout (code or XAML) which demonstrates this?
> 
> I've tried the following layout to reproduce this problem, but both sets of labels lay out identically:
> 
> ```
> <StackLayout>
> 
> 	<StackLayout Orientation="Horizontal" >
> 
> 		<Label Text="Label 1" />
> 		<Label Text="Label 2" />
> 		<Label Text="Label 3" />
> 		<Label Text="Label 4" />
> 		<Label Text="Label 5" />
> 		<Label Text="Label 6" />
> 		<Label Text="Label 7" />
> 		<Label Text="Label 8" />
> 
> 	</StackLayout>
> 
> 	<StackLayout Orientation="Horizontal" >
> 
> 		<Label Text="Label 1" AutomationId="label1" />
> 		<Label Text="Label 2" AutomationId="label2" />
> 		<Label Text="Label 3" AutomationId="label3" />
> 		<Label Text="Label 4" AutomationId="label4" />
> 		<Label Text="Label 5" AutomationId="label5" />
> 		<Label Text="Label 6" AutomationId="label6" />
> 		<Label Text="Label 7" AutomationId="label7" />
> 		<Label Text="Label 8" AutomationId="label8" />
> 
> 	</StackLayout>
> 
> </StackLayout>
> ```
> Am I missing a step to reproduce this?

Add helptext to each of those labels, add for example "tap here to access ..........many things ", add a long string to each of those labels. so they will all expand according to the size of the string.

**Comment by raymondjstone** [comment]
It's meant to be fixed on 3.5 and above I believe but I have not had a chance to check yet HOWEVER the issue only happens when the accessibility help text is longer than the original label text and your example above is not setting that and so should not have the issue. This is a different field to the Automation ID field
